File: models/src/main/java/com/mdgd/pokemon/models/repo/dao/schemas/MoveFullSchema.java
package com.mdgd.pokemon.models.repo.dao.schemas;

import androidx.room.Embedded;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import com.mdgd.pokemon.models.repo.schemas.VersionGroupDetail;

import java.util.ArrayList;
import java.util.List;

public class MoveFullSchema {

    @Embedded
    private MoveSchema move;

    @Expose
    @SerializedName("version_group_details")
    private List<VersionGroupDetail> versionGroupDetails = new ArrayList<>();

    public MoveSchema getMove() {
        return move;
    }

    public void setMove(MoveSchema move) {
        this.move = move;
    }

    public List<VersionGroupDetail> getVersionGroupDetails() {
        return versionGroupDetails;
    }

    public void setVersionGroupDetails(List<VersionGroupDetail> versionGroupDetails) {
        this.versionGroupDetails = versionGroupDetails;
    }
}

File: models_impl/src/main/java/com/mdgd/pokemon/models/repo/dao/schemas/MoveFullSchema.kt
package com.mdgd.pokemon.models.repo.dao.schemas

import androidx.room.Embedded
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.mdgd.pokemon.models.repo.schemas.VersionGroupDetail
import java.util.*

class MoveFullSchema {
    @Embedded
    var move: MoveSchema? = null

    @Expose
    @SerializedName("version_group_details")
    var versionGroupDetails: MutableList<VersionGroupDetail?> = ArrayList()
}
